### ISSUE TITLE
Remove stale sidebar entries for deleted accounts 

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#5756]
### Actual behavior
Historically, only the default local sync folder suggested by the wizard (`$USER/$APPLICATION_NAME`) got a sidebar link entry on finder/explorer. 

After introducing https://github.com/owncloud/client/issues/5597, when configuring multiple accounts you end up with multiple sidebar links; that's expected behavior. 

However, when removing an account from the client, the **links** stay there. (e.g. Win7 after https://github.com/owncloud/client/issues/2719)

<p align="center">
  <img src="https://cloud.githubusercontent.com/assets/2644445/25802864/99302ab4-33f4-11e7-9caf-b9ec378093ba.png"/>
</p>

### Expected behavior
Delete those entries that reference deleted accounts from the file explorer's sidebar. 

### Steps to reproduce
1. Add multiple accounts on the client **using the default folder name** suggested in the wizard 
2. Remove all the accounts created

**Comment by guruz** [comment]
Do you think this was always the case or is a regression from https://github.com/owncloud/client/pull/5690 ?

FYI @remixtj

**Comment by remixtj** [comment]
As far as i know there no favorite deletion procedure

**Comment by SamuAlfageme** [comment]
@guruz nope, nothing to do with #5690, this issue is cross-platform. It happens because now we have consecutive **default** suggested folders on the account setup wizard which get passed as argument to [`Utility::setupFavLink();`](https://github.com/owncloud/client/blob/master/src/gui/owncloudsetupwizard.cpp#L394). **If you manually create/use a local sync folder, no link is created on the sidebar. **
